fgOpenGL/BackendGL.cpp

```diff
@@ -142,23 +142,108 @@ int Backend::Clear(FG_Backend* self, void* commands, uint8_t clearbits, FG_Color
 
   return ERR_NOT_IMPLEMENTED;
 }
-int Backend::CopyResource(FG_Backend* self, void* commands, FG_Resource src, FG_Resource dest)
+int Backend::CopyResource(FG_Backend* self, void* commands, FG_Resource src, FG_Resource dest, FG_Vec3i size, int level)
 {
   auto backend = static_cast<Backend*>(self);
-  return ERR_NOT_IMPLEMENTED;
+
+  GLuint source = src;
+  GLuint destination = dest;
+
+  if (IsBuffer(source) && IsBuffer(dest)) 
+  {
+    return CopySubresource(self, commands, src, dest, 0, 0, size.x);
+  }
+  else if((IsTexture(source) && IsTexture(destination)) || (IsRenderbuffer(source) && IsRenderbuffer(destination)))
+  {
+    return CopyResourceRegion(self, commands, src, dest, level, FG_Vec3i{ 0, 0, 0 }, FG_Vec3i{ 0, 0, 0 },
+                              FG_Vec3i{ size.x, size.y, size.z });
+  }
+  else
+  {
+    backend->LOG(FG_Level_Error, "Mismatched src / dest resources");
+    return 0;
```

Comment:
Remember that 0 means success, anything else is a failure. It helps to use the built-in enums so you don't have to remember - you can return ERR_INVALID_PARAMETER here and ERR_SUCCESS below. 

This whole "else" statement is also not necessary, as you return from the previous if statements, so you can simplify this by removing the else and just having it immediately log a mismatch error.